src/libraries/System.Numerics.Tensors/tests/TensorPrimitivesTests.cs

```diff
@@ -57,6 +57,66 @@ private static unsafe float MathFMinMagnitude(float x, float y)
             float ax = MathF.Abs(x), ay = MathF.Abs(y);
             return (ax < ay) || float.IsNaN(ax) || (ax == ay && *(int*)&x < 0) ? x : y;
         }
+
+        private static unsafe int SingleToInt32(float f) => *(int*)&f;
+
+        private static unsafe float Int32ToSingle(int i) => *(float*)&i;
+
+        private static float AnotherSingleNaN = Int32ToSingle(-8388607);
+
+        /// <summary>Loads a variety of special values (e.g. NaN) into random positions in <paramref name="x"/>.</summary>
+        private static void SetSpecialValues(Span<float> x)
+        {
+            // NaN
+            x[s_random.Next(x.Length)] = float.NaN;
```

Comment:
Do we have any concern around `s_random.Next()` overwriting a previous random?

Reply:
My thinking is that we're relying on this being used enough that we'll have good enough coverage of everything.  Alternatively, we could test out all combinations.  I've been thinking we start with the former and graduate to the latter if/when we want the 100% coverage.